### PR TITLE
Update module:

### PR DESCRIPTION
**Pull request by cstamas**
* drop m-shared-utils (was used only for StringUtils!!!)
* drop plexus container default and plexus annotations, migrate to JSR330
* set plexus-classworld scope proper, IS USED in main

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -77,46 +77,42 @@
...
-      <groupId>org.codehaus.plexus</groupId>
-      <artifactId>plexus-component-annotations</artifactId>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-utils</artifactId>
-      <version>3.1.0</version><!-- do not upgrade to 3.2.0+ because it adds Xpp3Dom.getInputLocation(): see MSHARED-921 -->
+      <version>3.4.2</version>
     </dependency>
```
Comment (michael-o) [inline pom.xml:116]: I'd like to have Hervé's consent on this.

File: pom.xml [hunk 2]
```diff
@@ -188,11 +190,11 @@
       <scope>test</scope>
     </dependency>
 
-    <!-- Used actually in tests only, but if in test scope the transitive dep will be gone -->
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-classworlds</artifactId>
-      <version>2.5.2</version>
+      <version>2.6.0</version>
+      <scope>compile</scope>
```
Comment (michael-o) [inline pom.xml:197]: Do we really need this in compile?

File: pom.xml (hunk 1 above)
Comment (michael-o) [reply pom.xml:116]: Then you also need to remove the shading

File: pom.xml (hunk 2 above)
Comment (cstamas) [reply pom.xml:197]: Comment is totally wrong, and it WAS in compile scope already, I just made it obvious.

File: pom.xml (hunk 1 above)
Comment (michael-o) [reply pom.xml:116]: @hboutemy WDYT? For me, all MSITE ITs are working. I am fine with this.

**Review by michael-o — CHANGES_REQUESTED** [review]
While the `provided` scope looks fine to me, I'd like to keep it with the other components: `-impl` and move the scope change to the next milestone for consistency reasons.

BTW: All ITs pass for me.

**Comment by hboutemy** [comment]
from https://issues.apache.org/jira/browse/MSHARED-921 , what is important is to check with older Maven versions ( < 3.6.1 ): @michael-o  did you try with Maven 3.3.9?

File: pom.xml (hunk 1 above)
Comment (hboutemy) [reply pom.xml:116]: from https://issues.apache.org/jira/browse/MSHARED-921 , what is important is to check with older Maven versions ( < 3.6.1 ): @michael-o did you try with Maven 3.3.9?

**Comment by michael-o** [comment]
> from https://issues.apache.org/jira/browse/MSHARED-921 , what is important is to check with older Maven versions ( < 3.6.1 ): @michael-o did you try with Maven 3.3.9?

I actually did, but want to double-check. exec is used in reporting plugins only, means no maven-site-plugin, direct invocation. Therefore, I'd would be good enough to run a few of them with this change and when they work then this is fine?

**Comment by michael-o** [comment]
@hboutemy  Testing again

**Comment by michael-o** [comment]
I installed exec from this branch. Modified MSITE to use this SNAPSHOT, used Maven 3.3.9 to install and run ITs. Works. Used Doxia 2 branches on MPIR and MPLUGIN, updated MSITE to M3-SNAPSHOT, again with 3.3.9 install and ITs. Works.

@hboutemy Is that sufficient?

**Comment by hboutemy** [comment]
I need to check Git history to understand

> exec is used in reporting plugins only, means no maven-site-plugin, direct invocation.

no, exec is used in m-site-p and m-pdf-p, to run reports

**Comment by michael-o** [comment]
> I need to check Git history to understand
> 
> > exec is used in reporting plugins only, means no maven-site-plugin, direct invocation.
> 
> no, exec is used in m-site-p and m-pdf-p, to run reports

You are totally right. My statement is wrong. I messed it up with `-impl`.

**Comment by hboutemy** [comment]
I really don't understand why Xpp3DomUtils.mergeXpp3Dom call works now with older Maven 3.3.9 without the shade trick https://github.com/apache/maven-reporting-exec/blob/master/src/main/java/org/apache/maven/reporting/exec/DefaultMavenReportExecutor.java#L514 

but it seems to just work now... (if anybody can understand, please help :) )

ok for me to merge

**Comment by michael-o** [comment]
> I really don't understand why Xpp3DomUtils.mergeXpp3Dom call works now with older Maven 3.3.9 without the shade trick https://github.com/apache/maven-reporting-exec/blob/master/src/main/java/org/apache/maven/reporting/exec/DefaultMavenReportExecutor.java#L514
> 
> but it seems to just work now... (if anybody can understand, please help :) )
> 
> ok for me to merge

Yepp, this is what I am trying to understand as well...

**Comment by michael-o** [comment]
@hboutemy So I assume we have your consent?

**Comment by cstamas** [comment]
@michael-o pls take over this PR, as I don't think will have time to finish it

**Pull request merged**